src/Nethermind/Nethermind.Core2/Types/Slot.cs

```diff
@@ -124,6 +124,11 @@ public override int GetHashCode()
         {
             return slot.Number;
         }
+        
+        public static implicit operator int(Slot slot)
```

Comment:
I worry about direct conversion to int, either implicit or explicit, as the Eth2 spec is that these values are uint64. 

I know that .NET arrays are constrained to int32 indexes, and maybe we will never reach slot numbers that big, but a better approach may be explicit accessors (GetXxx) that check values.